### DISCUSSION TITLE
What is the magic code for the pipe symbol? It looks like this: |

### DISCUSSION THREAD

**Discussion by byron-hawkins** [sezanzeb/input-remapper#87] [Q&A]
I searched the whole list of `KEY_XXX` from A to Z and did not find anything called `KEY_PIPE` or `KEY_BAR` or any such intuitive name for the '|`symbol that is missing from my 101-key in UK layout. Is there a visual table somewhere that correlates the magic names with the visual glyph that appears on the screen when entered (or meta command details if not a glyph key)? The linux documentation only gives corelation between key codes and names, with no indication what glyph goes with those arbitrarily selected names.

**Comment by sezanzeb** [comment]
See https://github.com/sezanzeb/key-mapper/blob/main/readme/usage.md#how-to-use-unavailable-characters and use `bar`

are you only seeing names starting with "KEY_"? If yes, what is the output of `xmodmap -pke` then?

**Reply by byron-hawkins** [reply]
The list appears to be complete, but evidently, by design it only includes the "plain" version of each key. So symbols like pipe and question mark and angle brackets are only reachable by shift combinations. That means guessing the "standard" position of those keys, or testing all the shift combos by trial and error (or with an app if there is one), since they cannot be referenced directly by name. But of course the pipe is on `SHIFT_L + backslash` so at least my problem is solved for now. 

Thanks for the great tool, this has been a major obstacle for me on linux for years, because even editing xmodmap is not reliable without very specific knowledge of the service startup sequence (key mappings become mysteriously disabled in various scenarios).

**Reply by sezanzeb** [reply]
Thanks! I'm glad it helps

I once included symbols like "bar" in the list, but since without shift it ends up writing a totally different symbol (like you figured out) which looked more like a bug than a feature. So I removed that.

**Reply by sezanzeb** [reply]
https://github.com/sezanzeb/key-mapper/issues/90
